api/src/main/java/org/openmrs/module/m2sysbiometrics/model/M2SysResults.java

```diff
@@ -49,6 +49,10 @@ public boolean isChangeIdSuccess() {
         return firstValueEqualsIgnoreCase(M2SysResult.UPDATE_SUBJECT_ID_SUCCESS);
     }
 
+    public boolean isSearchError() {
+        return results.get(0).getScore() == M2SysResult.SEARCH_ERROR;
```

Comment:
If there are no hits the response is:

<Results><result score="0" value="-1" Instance="2"/></Results>

which in this case will improperly be categorized as error. You need to look at value as well

Reply:
You should also check for empty results, like firstValue() does

Reply:
ok i will